### PR TITLE
Fix spirvreader image/sampler translating failure

### PR DESCRIPTION
**Pull request by jiaolu**
1. Image stride is translated to int32 type. so image descriptor index should be cast to the int32 type.
2. Image is translated to the struct/array type. so it can not be applyed to the recursive load/store operation.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit 91bb57aca2ccfce2d0487444b1ba0348d99a1dd0
<details><summary>CTS tests (Failed: 0/211258)</summary><ul><li>Built with version 1.3.3.1</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 39024/70423 (55.4%)</li>
<li>Failed: 0/70423 (0.0%)</li>
<li>Not Supported: 31399/70423 (44.6%)</li><li>Warnings: 0/70423 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 39058/70410 (55.5%)</li>
<li>Failed: 0/70410 (0.0%)</li>
<li>Not Supported: 31352/70410 (44.5%)</li><li>Warnings: 0/70410 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 35147/70425 (49.9%)</li>
<li>Failed: 0/70425 (0.0%)</li>
<li>Not Supported: 35278/70425 (50.1%)</li><li>Warnings: 0/70425 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by jiaolu** [comment]
ping for review.

**Review by nhaehnle — COMMENTED** [review]
I don't quite understand what is happening here. Do you have a sample pipeline / lit test case that is affected by the change?

**Comment by jiaolu** [comment]
> I don't quite understand what is happening here. Do you have a sample pipeline / lit test case that is affected by the change?

Lit test is added , the glsl is showing the intention of this pr. 

; layout(set = 0, binding = 7) uniform sampler2D _11[];
; layout(location = 0) out vec4 _3;
; layout(location = 1) flat in u64vec2 _4;
; void main()
; {
;     uint64_t _12 = _4.x;
;     sampler2D s = _11[nonuniformEXT(_12)];
;     _3 = texture(s, vec2(0.0, 0.0));
; }

**Comment by jiaolu** [comment]
retest this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit 18b94b740878823e60f021bc36b5afeb18366345
<details><summary>CTS tests (Failed: 197/211353)</summary><ul><li>Built with version 1.3.3.1</li>
<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 35125/70456 (49.9%)</li>
<li><details><summary>Failed: 22/70456 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.composites.10
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.composites.2
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.composites.3
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.bool
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.f16mat2x2
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.f16mat3x2
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.f16mat3x4
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.f32mat2x2
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.f32mat2x3
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.f32mat2x4
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.f32mat4x3
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.f64mat2x3
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.f64vec3
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.i64vec3
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.int16_t
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.int64_t
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.ivec4
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.u8vec2
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.uint16_t
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.uint32_t
Stack trace: Crash
...
```
</p></details></li>

<li>Not Supported: 35309/70456 (50.1%)</li><li>Warnings: 0/70456 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 38873/70456 (55.2%)</li>
<li><details><summary>Failed: 153/70456 (0.2%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.repeat_pipeline.x_16_odd_repeat_2
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.repeat_pipeline.x_32_odd_repeat_16
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.repeat_pipeline.x_32_odd_repeat_2
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.repeat_pipeline.x_4_even_repeat_4
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.repeat_pipeline.x_4_odd_repeat_2
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.repeat_pipeline.x_64_even_repeat_16
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.repeat_pipeline.x_64_even_repeat_2
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.repeat_pipeline.x_64_odd_repeat_16
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.specialize_workgroup.1_1_3
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.specialize_workgroup.1_2_3
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.specialize_workgroup.1_2_7
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.specialize_workgroup.1_3_1
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.specialize_workgroup.1_3_3
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.specialize_workgroup.1_3_4
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.specialize_workgroup.1_3_5
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.specialize_workgroup.1_4_3
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.specialize_workgroup.1_4_7
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.specialize_workgroup.1_5_3
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.specialize_workgroup.1_5_5
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.specialize_workgroup.1_6_1
Stack trace: Crash
...
```
</p></details></li>

<li>Not Supported: 31430/70456 (44.6%)</li><li>Warnings: 0/70456 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 39036/70441 (55.4%)</li>
<li><details><summary>Failed: 22/70441 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.composites.10
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.composites.2
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.composites.3
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.bool
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.f16mat2x2
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.f16mat3x2
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.f16mat3x4
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.f32mat2x2
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.f32mat2x3
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.f32mat2x4
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.f32mat4x3
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.f64mat2x3
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.f64vec3
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.i64vec3
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.int16_t
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.int64_t
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.ivec4
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.u8vec2
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.uint16_t
Stack trace: Crash

FAILURE: dEQP-VK.compute.zero_initialize_workgroup_memory.types.uint32_t
Stack trace: Crash
...
```
</p></details></li>

<li>Not Supported: 31383/70441 (44.6%)</li><li>Warnings: 0/70441 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by qiaojbao** [comment]
these crash should be fixed in latest dev branch, try to rebase

**Comment by amdvlk-admin** [comment]
### Test summary for commit ba19b5f8706bae8d7e46910ac66c3b557c60a7b7
<details><summary>CTS tests (Failed: 0/211353)</summary><ul><li>Built with version 1.3.3.1</li>
<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 35147/70456 (49.9%)</li>
<li>Failed: 0/70456 (0.0%)</li>
<li>Not Supported: 35309/70456 (50.1%)</li><li>Warnings: 0/70456 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 39026/70456 (55.4%)</li>
<li>Failed: 0/70456 (0.0%)</li>
<li>Not Supported: 31430/70456 (44.6%)</li><li>Warnings: 0/70456 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 39058/70441 (55.4%)</li>
<li>Failed: 0/70441 (0.0%)</li>
<li>Not Supported: 31383/70441 (44.6%)</li><li>Warnings: 0/70441 (0.0%)</li>
</ul></details>


</ul></details>

File: llpc/test/shaderdb/core/ObjNonUniform_TestTexutreLoadStoreInt64.spvasm [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+         %26 = OpCopyObject %uint %25
+         %28 = OpAccessChain %_ptr_UniformConstant_21 %_11 %26
+         %29 = OpLoad %21 %28
+               OpStore %texture %29
+  %loadImage = OpLoad %21 %texture
+         %33 = OpImageSampleImplicitLod %v4float %loadImage %32
+               OpStore %_3 %33
+               OpReturn
+               OpFunctionEnd
+; CHECK-LABEL: @main(
```
Comment (nhaehnle) [inline llpc/test/shaderdb/core/ObjNonUniform_TestTexutreLoadStoreInt64.spvasm:89]: The CHECK- lines can be removed, right? I only see the one RUN line that uses SHADERTEST

**Review by nhaehnle — COMMENTED** [review]
This general area of code is hard to follow, but I think this change is okay.

**Comment by jiaolu** [comment]
> This general area of code is hard to follow, but I think this change is o

**Comment by amdvlk-admin** [comment]
### Test summary for commit 8f33f04a960b26a1a7af9870e4d5299c40552c33
<details><summary>CTS tests (Failed: 0/211353)</summary><ul><li>Built with version 1.3.3.1</li>
<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 35147/70456 (49.9%)</li>
<li>Failed: 0/70456 (0.0%)</li>
<li>Not Supported: 35309/70456 (50.1%)</li><li>Warnings: 0/70456 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 39026/70456 (55.4%)</li>
<li>Failed: 0/70456 (0.0%)</li>
<li>Not Supported: 31430/70456 (44.6%)</li><li>Warnings: 0/70456 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 39058/70441 (55.4%)</li>
<li>Failed: 0/70441 (0.0%)</li>
<li>Not Supported: 31383/70441 (44.6%)</li><li>Warnings: 0/70441 (0.0%)</li>
</ul></details>


</ul></details>

**Review by nhaehnle — APPROVED** [review]
